flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/StateTransition.java

```diff
@@ -36,24 +34,14 @@
 	private final StateTransitionAction action;
 	private final State<T> sourceState;
 	private final State<T> targetState;
-	private IterativeCondition<T> newCondition;
-
-	/**
-	 * @deprecated 	This field remains for backwards compatibility.
-	 * Now the conditions extend the {@link IterativeCondition}.
-	 */
-	@Deprecated
-	private FilterFunction<T> condition;
```

Comment:
I think this field needs to stay, though. It was prior to 1.3.x, so it is needed for java serialization, which is used for snapshots taken in 1.2.x